File: hello_mpi.cpp
#include <mpi.h>
#include <iostream>

int main(int argc, char** argv) {
    MPI_Init(&argc, &argv);

    int world_size;
    MPI_Comm_size(MPI_COMM_WORLD, &world_size);

    int world_rank;
    MPI_Comm_rank(MPI_COMM_WORLD, &world_rank);

    std::cout << "Hello from rank " << world_rank << " out of " << world_size << " processors" << std::endl;

    MPI_Finalize();
    return 0;
}

File: hello_mpi.f90
program mpi_hello
    use mpi
    implicit none
    integer :: ierr, rank, size

    call MPI_Init(ierr)
    call MPI_Comm_size(MPI_COMM_WORLD, size, ierr)
    call MPI_Comm_rank(MPI_COMM_WORLD, rank, ierr)

    print *, 'Hello from rank', rank, 'of', size, 'processors'
    call MPI_Finalize(ierr)
end program mpi_hello
